server/neptune/workflows/internal/activities/deployment/store.go

```diff
@@ -0,0 +1,80 @@
+package deployment
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/neptune/storage"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/root"
+)
+
+const OutputPrefix = "deployments"
+
+type client interface {
+	Get(ctx context.Context, key string) (io.ReadCloser, error)
+	Set(ctx context.Context, key string, object []byte) error
+}
+
+func NewStore(stowClient client) (*store, error) {
+	return &store{
+		stowClient: stowClient,
+	}, nil
+
+}
+
+type store struct {
+	stowClient client
+}
+
+func (s *store) GetDeploymentInfo(ctx context.Context, repoName string, rootName string) (*root.DeploymentInfo, error) {
+	key := BuildKey(repoName, rootName)
+
+	reader, err := s.stowClient.Get(ctx, key)
+	if err != nil {
+		switch err.(type) {
+
+		// Fail if container is not found
+		case *storage.ContainerNotFoundError:
+			return nil, err
+
+		// First deploy for this root
+		case *storage.ItemNotFoundError:
+			return &root.DeploymentInfo{}, nil
```

Comment:
What's the point in returning a pointer to an empty struct, should we just return a nil pointer and check for it upstream?